### ISSUE TITLE
Bootstrap GSM tech debt

### ISSUE THREAD

**Issue by nfrisby** [IntersectMBO/ouroboros-consensus#964]
Taken from https://github.com/IntersectMBO/ouroboros-consensus/pull/808#issuecomment-1954913114, upon clicking the "Merge when ready" button.

- [ ] s-r-p stanzas (see https://github.com/IntersectMBO/ouroboros-consensus/pull/808#discussion_r1490714381)
- [ ] move the anti-thundering herd delay into the Diffusion Layer, _between_ the two states CaughtUp and OnlyBootstrap
- regression tests (code review and ad-hoc manual testing has already evidenced that they're _currently_ correct):
    - [ ] `realMarkerFileView` regression tests
    - [ ] `initializationLedgerJudgement` regression tests
    - [ ] `realDurationUntilTooOld` regression tests
    - [ ] The ChainSync client should signal idle exactly when and only when the `MsgAwaitReply` is processed.
    - [ ] The ChainSync client subsequently should signal no-longer-idle exactly when and only when the next message is received.
- [ ] switch to `quickcheck-dynamic` and/or upstream the `io-sim`-based QSM module
- [ ] use `ArbitraryLedgerStateJudgement` from `ouroboros-network:sim-tests-lib`

**Comment by jasagredo** [comment]
- Either move the GSM logic into the ChainDB and keep the GSM arguments as part of the ChainDB arguments (which would be encapsulated inside of `openChainDB`) or remove the argument from the ChainDB arguments and keep it as a NodeKernel-level thing.

**Comment by amesgen** [comment]
Yes, the GSM now actually deserves its name. The changes for that were rather small:

 - https://github.com/IntersectMBO/ouroboros-consensus/pull/975
 - https://github.com/IntersectMBO/ouroboros-consensus/pull/1031

The tasks in this ticket should be independent of that.
